tests/core/types/test_account.py

```diff
@@ -0,0 +1,330 @@
+@patch("nile.core.types.account.Account._process_arguments")
+async def test_deploy(
+    mock_process_arguments,
+    mock_address,
+    mock_class_hash,
+    salt,
+    max_fee,
+    abi,
+):
+    account = await MockAccount(KEY, NETWORK)
+    calldata = [account.signer.public_key]
+    mock_process_arguments.return_value = ((max_fee or 0), 0, calldata)
+
+    with patch(
+        "nile.core.types.transactions.DeployAccountTransaction._get_tx_hash"
+    ) as mock_get_tx_hash:
+        mock_get_tx_hash.return_value = 0x777
+
+        tx_wrapper = await account.deploy(
+            salt=salt,
+            max_fee=max_fee,
+            abi=abi,
+        )
+
+        # Check transaction wrapper
+        assert tx_wrapper.account == account
+        assert tx_wrapper.alias == account.alias
+        assert tx_wrapper.contract_name == "Account"
+        assert tx_wrapper.predicted_address == MOCK_ADDRESS
+        assert tx_wrapper.overriding_path == NILE_ARTIFACTS_PATH
+        assert tx_wrapper.abi == abi
+
+        # Check transaction
+        tx = tx_wrapper.tx
+        assert tx.tx_type == "deploy_account"
+        assert tx.account_address == 0
+        assert tx.contract_to_submit == "Account"
+        assert tx.predicted_address == MOCK_ADDRESS
+        assert tx.calldata == calldata
+        assert tx.class_hash == CLASS_HASH
+        assert tx.overriding_path == NILE_ARTIFACTS_PATH
+        assert tx.max_fee == (max_fee or 0)
+        assert tx.nonce == 0
+        assert tx.network == account.network
+        assert tx.version == TRANSACTION_VERSION
+
+        # Check internals
+        mock_class_hash.assert_called_once_with(
+            contract_name="Account", overriding_path=NILE_ARTIFACTS_PATH
+        )
+
+        # Check '_process_arguments' call
+        mock_process_arguments.assert_called_once_with(max_fee, 0, calldata)
+
+
+@pytest.mark.asyncio
+@patch(
+    "nile.core.types.transactions.Transaction.execute",
+    return_value=(TX_STATUS, "output"),
+)
+@patch("nile.core.types.account.get_counterfactual_address", return_value=MOCK_ADDRESS)
+@patch("nile.core.types.transactions.get_class_hash", return_value=CLASS_HASH)
+@patch("nile.core.deploy.accounts.register")
+async def test_deploy_accounts_register(
+    mock_register, mock_hash, mock_address, mock_deploy
+):
+    account = await Account(KEY, NETWORK)
+
+    mock_register.assert_called_once_with(
+        account.signer.public_key, MOCK_ADDRESS, MOCK_INDEX, KEY, NETWORK
+    )
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("contract_name", ["contract"])
+@pytest.mark.parametrize("max_fee", [0, 15, None])
+@pytest.mark.parametrize("nonce", [0])
+@pytest.mark.parametrize("alias", ["my_contract"])
+@pytest.mark.parametrize("overriding_path", [(BUILD_DIRECTORY, ABIS_DIRECTORY), None])
+@patch("nile.core.types.transactions.get_contract_class", return_value="ContractClass")
+@patch("nile.core.types.account.Account._process_arguments")
+async def test_declare(
```

Comment:
I'm going to update the `Try the --stack_trace option for the full stack trace.` to `Try nile --stack_trace command for the full stack trace.` because I get that is not clear at all why that the option must be placed after nile.